homeassistant/components/zwave_js/diagnostics.py

```diff
@@ -17,9 +20,49 @@
 from homeassistant.helpers.entity_registry import async_entries_for_device, async_get
 
 from .const import DATA_CLIENT, DOMAIN
-from .helpers import get_home_and_node_id_from_device_entry
+from .helpers import ZwaveValueID, get_home_and_node_id_from_device_entry
 
-TO_REDACT = {"homeId", "location"}
+KEYS_TO_REDACT = {"homeId", "location"}
+
+VALUES_TO_REDACT = (
+    ZwaveValueID(property_="userCode", command_class=CommandClass.USER_CODE),
+)
+
+
+def redact_value_of_zwave_value(zwave_value: ValueDataType) -> ValueDataType:
+    """Redact value of a Z-Wave value."""
+    for value_to_redact in VALUES_TO_REDACT:
+        if (
+            (
+                value_to_redact.command_class is None
```

Comment:
Maybe create another `ZwaveValueID` instance out of the `zwave_value` and compare the dataclass instances directly. They have a equality magic method by default.

https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass

Reply:
Although I guess you only want to match on the attributes that are not None in the matcher.

Reply:
Maybe just iterate over fields of the matcher instance and check if not None and equal?

Reply:
that doesn't provide values so I am using `astuple` instead